text/0711-deprecate-auto-location.md

```diff
@@ -0,0 +1,140 @@
+---
+Stage: Accepted
+Start Date: 2021-01-23
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js
+RFC PR: https://github.com/emberjs/rfcs/pull/711
+---
+
+<!---
+Directions for above:
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# <RFC title>
+
+## Summary
+
+Deprecate Ember.AutoLocation and related APIs. In next major version,
+make history the default location (instead of auto).
+
+## Motivation
+
+In practice, 'auto' will almost always resolve to the 'history' location.
+
+By removing the 'auto' location and setting 'history' as the default we're removing
+some complexity around router location, with no downside for users, since they
+all get it resolved to history anyway.
+
+## Background
+
+The Ember Router can use different mechanisms to serialize its state.
+The two common ones are 'history' and 'hash'. There is also a special
+location called 'auto' which does feature detection, preferring 'history'.
+
+Basically all browsers will get 'history':
+https://caniuse.com/mdn-api_history_pushstate
```

Comment:
Might add some details about _why_ this exists at all (when we added AutoLocation, and made it default all of our browsers **didn't** support it).